src/types/tx.ts

```diff
@@ -31,9 +31,9 @@ export type TxResult = TxError | TxSuccess;
 
 // //////////// ESTIMATE TX ////////////
 export type TxContent =
-  | { type: "cosmos"; val: EncodeObject[] }
+  | { type: "cosmos"; val: EncodeObject[]; attribute?: string }
```

Comment:
I know the current setup "works fine at the moment", but it forces us to:
- pass either `attribute` or `log` parameter to `TxContent` when we need some param read from the logs, the purpose of which is not immediately obvious unless you know how it will be used in the subsequent chain of functions (to return `TxSuccess.attrValue`)
- have different ways of populating `TxSuccess.attrValue` (for `cosmos` we pass `attribute` and it finds it for us, while for `evm` we pass the log filter function that the `sendTx` has to call)
- return `TxSuccess.attrValue`, the purpose of which is not immediately obvious (unless you know how it was populated)
- might not be a need for this atm, but this setup would stop us from being able to read multiple attributes from the logs

What do you think about just returning `TxSuccess.logs` instead of returning `TxSuccess.attrValue` and processing those logs as we see fit in specific functions/components? `.

Reply:
> pass either attribute or log parameter to TxContent when we need to read some param from the logs,

You are right it's confusing. I'm still looking for ways to make it more comprehensible. The reason `attribute` and `fn log` is passed in `TxContent` rather than `sendTx` is that in `tx creation` step, this is where we distinguish a tx type from another along with how those differences manifest. E.g. for cosmos (`attribute`) and for polygon (`log`)


> have different ways of populating TxSuccess.attrValue (for cosmos we pass attribute and it finds it for us, while for evm we pass the log filter function that the sendTx has to call)

the reason they're different was due to the difference on how those logs can be parsed. For `cosmos` it's a simple finding `value` for a `key` (`attribute` which is human readable). which is typically just primitive `string | number`. For polygon or `evm` in general. `keys` are encoded based on `Event signature` and values are also encoded and needs to be `decoded` depending on the `Event signature`. Moreover, the `encoded` value isn't just some `string | number` an could be complex depending on `Event signature` 
sample
```
"event SwapToken(uint256,uint8,uint256,address,address,uint256)",
"event EndowmentSettingUpdated(uint256,string)",
```